receiver/otlpreceiver/otlp.go

```diff
@@ -77,11 +77,18 @@ func New(
 		return nil, fmt.Errorf("failed to bind to address %q: %v", addr, err)
 	}
 
+	protobufMarshaler := &xProtobufMarshaler{}
 	r := &Receiver{
 		ln:          ln,
 		corsOrigins: []string{}, // Disable CORS by default.
 		gatewayMux: gatewayruntime.NewServeMux(
-			gatewayruntime.WithMarshalerOption("application/x-protobuf", &xProtobufMarshaler{}),
+			gatewayruntime.WithMarshalerOption("application/x-protobuf", protobufMarshaler),
+			// Explicitly handle "application/json" encoding to return custom human friendly error
+			// See issue https://github.com/open-telemetry/opentelemetry-proto/issues/129
+			// describing why we don't support JSON at the moment.
+			gatewayruntime.WithMarshalerOption("application/json", newJSONUnsupportedMarshaler()),
+			// Override default JSON handling with Protobuf
+			gatewayruntime.WithMarshalerOption(gatewayruntime.MIMEWildcard, protobufMarshaler),
```

Comment:
This is the main change in this PR - by default, gRPC Gateway is using JSON marshaler for any content type other than explicitly configured above. Here I changed it to use Protobuf marshaler by default.

Alternatively, we may require always setting `Content-Type: application/x-protobuf` explicitly and return an error for `*` wildcard here.